---
title: "Yeast Data QC"
output: html_notebook
---



```{r library, include=FALSE}

library(brentlabRnaSeqTools)
library(tidyverse)
library(ggExtra)
library(here)

# broad_rnaseqc_output = "/mnt/htcf_scratch/chasem/rnaseq_pipeline/yeast_results/broad_rnaseqc_output/"
# 
# multiqc_general_stats_path = "/mnt/htcf_scratch/chasem/rnaseq_pipeline/yeast_results/multiqc/star_salmon/all_yeast_samples_multiqc_report_data/multiqc_general_stats.txt"

multiqc_general_stats = read_csv(here('data/multiqc_general_stats.csv'))

multiqc_general_stats = multiqc_general_stats %>%
  dplyr::rename(sample = Sample) %>%
  dplyr::select(sample,
                `Samtools_mqc-generalstats-samtools-flagstat_total`,
                `Samtools_mqc-generalstats-samtools-mapped_passed`,
                `Picard_mqc-generalstats-picard-PERCENT_DUPLICATION`,
                `DupRadar_mqc-generalstats-dupradar-dupRadar_intercept`,
                `Biotype Counts_mqc-generalstats-biotype_counts-percent_rRNA`,
                `QualiMap_mqc-generalstats-qualimap-5_3_bias`,
                `FastQC (trimmed)_mqc-generalstats-fastqc_trimmed-percent_duplicates`, 
                `FastQC (trimmed)_mqc-generalstats-fastqc_trimmed-percent_gc`)

```

dup intercept, see dupRadar https://bioconductor.org/packages/release/bioc/vignettes/dupRadar/inst/doc/dupRadar.html:

With the experience from the ENCODE datasets, we expect from single read experiments little duplication at low RPKM (low intercept) rapidly rising because of natural duplication (high slope). In contrast, paired-end experiments have a more mild rising of the natural duplication (low slope) due to having higher diversity of reads pairs since pairs with same start may still have different end.

The common denominator for both designs is the importance of having a low intercept, suggesting that duplication rate at lowly expressed genes may serve as a quality measure.

```{r}
# rnaseqc_output = parseBroadRnaseqcOutput(broad_rnaseqc_output)
# write_rds(rnaseqc_output, here("data/parsed_rnaseqc_output.rds"))
rnaseqc_output = readRDS(here("data/parsed_rnaseqc_output.rds"))

rnaseqc_multiqc = rnaseqc_output$rnaseqc_df_subset %>%
  left_join(multiqc_general_stats) %>%
  select(-c("rRna_rate", "estimated_library_complexity"))
```

```{r correlation}


rnaseqc_multiqc_cor = cor(select(rnaseqc_multiqc, -sample))

testRes = cor.mtest(select(rnaseqc_multiqc, -sample), conf.level=0.95)
# see https://cran.r-project.org/web/packages/corrplot/vignettes/corrplot-intro.html#visualize-p-value-and-confidence-interval
show(corrplot(rnaseqc_multiqc_cor,
              p.mat = testRes$p,
              sig.level = .10,
              method = 'square', 
              insig='blank',
              addCoef.col ='black', 
              number.cex = 0.8,
              order = 'FPC', 
              type = 'lower', 
              diag = FALSE))


```


```{r plot epe vs rRna}
rRna_vs_epe = rnaseqc_multiqc %>%
    ggplot(aes(`Biotype Counts_mqc-generalstats-biotype_counts-percent_rRNA`, expression_profiling_efficiency)) +
    geom_point() +
    scale_x_continuous() +
    scale_y_continuous()

rRna_vs_epe_with_marginals = ggMarginal(rRna_vs_epe, type="histogram")
```


```{r}
rRna_vs_epe_with_marginals
```

```{r plot duplicated percent vs rRna}
epe_vs_duplication = rnaseqc_multiqc %>%
    ggplot(aes(`Picard_mqc-generalstats-picard-PERCENT_DUPLICATION`, `expression_profiling_efficiency`)) +
    geom_point() +
    scale_x_continuous() +
    scale_y_continuous()

epe_vs_duplication_with_marginals = ggMarginal(rRna_vs_duplicationRate, type="histogram")
```

```{r}
epe_vs_duplication_with_marginals
```


```{r read placement histograms}

QualiMap_mqc-generalstats-qualimap-5_3_bias


```


```{r plot 5 3 bias against duplication rate}
duplication_vs_5_3_bias = rnaseqc_multiqc %>%
    ggplot(aes(`Picard_mqc-generalstats-picard-PERCENT_DUPLICATION`, `QualiMap_mqc-generalstats-qualimap-5_3_bias`)) +
    geom_point() +
    scale_x_continuous() +
    scale_y_continuous()

duplication_vs_5_3_bias_with_marginals = ggMarginal(duplication_vs_5_3_bias, type="histogram")
```

```{r}
duplication_vs_5_3_bias_with_marginals
```

```{r plot 5 3 bias against epe}
epe_vs_5_3_bias = rnaseqc_multiqc %>%
    ggplot(aes( `QualiMap_mqc-generalstats-qualimap-5_3_bias`, `expression_profiling_efficiency` )) +
    geom_point() +
    scale_x_continuous() +
    scale_y_continuous()

epe_vs_5_3_bias_with_marginals = ggMarginal(epe_vs_5_3_bias, type="histogram")
```

```{r}
epe_vs_5_3_bias_with_marginals
```

```{r plot duplicated percent vs rRna}
duplication_vs_5_3_bias = rnaseqc_multiqc %>%
    ggplot(aes(`Picard_mqc-generalstats-picard-PERCENT_DUPLICATION`, `QualiMap_mqc-generalstats-qualimap-5_3_bias`)) +
    geom_point() +
    scale_x_continuous() +
    scale_y_continuous()

duplication_vs_5_3_bias_with_marginals = ggMarginal(duplication_vs_5_3_bias, type="histogram")
```

```{r}
duplication_vs_5_3_bias_with_marginals
```

